### PR TITLE
houdini: improve packaging

### PR DESCRIPTION
**Pull request by permahorse**
## Description of changes

* fixed issues with wrappers
* exposed additional binaries
* restructured runtime.nix for easier updates in future

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onemoresuza** [comment]
Result of `nixpkgs-review pr 256685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>houdini</li>
  </ul>
</details>

**Comment by pedohorse** [comment]
> Result of `nixpkgs-review pr 256685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 
>     * houdini

can this be because the package uses `requireFile` ? the required installation file cannot be easily downlowded automatically unfortunately, you need to register an (free) account on sidefx.com to proceed to the download page

with required file added to nix store `nixpkgs-review pr 256685` works for me

**Comment by onemoresuza** [comment]
> > Result of `nixpkgs-review pr 256685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> > 1 package failed to build:
> > ```
> > * houdini
> > ```
> 
> can this be because the package uses `requireFile` ? the required installation file cannot be easily downlowded automatically unfortunately, you need to register an (free) account on sidefx.com to proceed to the download page
> 
> with required file added to nix store `nixpkgs-review pr 256685` works for me

It looks like it, the error it showed did not look like a, let's say, nixy problem.

Could you run `nixpkgs-review pr --post-result 256685` so that the result is posted to this PR?

**Comment by pedohorse** [comment]
> `nixpkgs-review pr --post-result 256685`

ran with `--print-result` and copy-pasted here, hope it's the same:

Result of `nixpkgs-review pr 256685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>houdini</li>
  </ul>
</details>

**Comment by onemoresuza** [comment]
Same thing.

**Review by onemoresuza — APPROVED** [review]

**Comment by pedohorse** [comment]
Thank you for approving the PR!
Is there someone I need to ping to merge it now?

**Comment by onemoresuza** [comment]
Let a few days past; if it is yet to be merged, try posting it again to
[PRs ready for review](https://discourse.nixos.org/t/prs-ready-for-review/3032) stating
that both the package has been approved and the `nixpkgs-review` succeed
in building it.

File: pkgs/applications/misc/houdini/runtime-build.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, requireFile, bc, version, src, eulaDate ? "2021-10-13"}:
```
[the remark this answers is not in the document]
Comment (pedohorse) [reply pkgs/applications/misc/houdini/runtime-build.nix]: done

File: pkgs/applications/misc/houdini/runtime.nix [hunk 2]
```diff
@@ -1,40 +1,13 @@
-{ lib, stdenv, requireFile, bc }:
+{ lib, stdenv, requireFile, callPackage}:
 
 let
   license_dir = "~/.config/houdini";
 in
-stdenv.mkDerivation rec {
+callPackage ./runtime-build.nix rec {
   version = "19.5.569";
```
[the remark this answers is not in the document]
Comment (pedohorse) [reply pkgs/applications/misc/houdini/runtime.nix:7]: and done

**Comment by pedohorse** [comment]
1 package built:
houdini

Result of `nixpkgs-review pr 256685` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>houdini</li>
  </ul>
</details>

**Pull request merged**